File: System/Workspace/Allors.Workspace/IWorkspace.cs
// <copyright file="IWorkspace.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace
{
    using System;
    using System.Collections.Generic;
    using Derivations;
    using Meta;

    public interface IWorkspace
    {
        string Name { get; }

        IMetaPopulation MetaPopulation { get; }

        IObjectFactory ObjectFactory { get; }

        IWorkspaceLifecycle StateLifecycle { get; }

        ISession CreateSession();

        IDictionary<Guid, IDomainDerivation> DomainDerivationById { get; }
    }
}

File: typescript/libs/workspace/domain/system/src/lib/IWorkspace.ts
import { MetaPopulation } from "@allors/workspace/meta/system";
import { IObjectFactory } from "./IObjectFactory";
import { ISession } from "./ISession";
import { IWorkspaceServices } from "./state/IWorkspaceServices";

export interface IWorkspace {
  name: string;

  metaPopulation: MetaPopulation;

  objectFactory: IObjectFactory;

  lifecycle: IWorkspaceServices;

  createSession(): ISession;
}
